cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -162,21 +163,21 @@ writeProtocolParameters mOutFile pparams =
         LBS.writeFile fpath (encodePretty pparams)
 
 runQueryTip
-  :: Protocol
+  :: AnyConsensusModeParams
   -> NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
-runQueryTip protocol network mOutFile = do
+runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
     SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
-    output <-
-      firstExceptT ShelleyQueryCmdLocalStateQueryError $
-      withlocalNodeConnectInfo protocol network sockPath $ \connectInfo -> do
-        tip <- liftIO $ getLocalTip connectInfo
-        let output = case localNodeConsensusMode connectInfo of
-                       ByronMode{}   -> encodePretty tip
-                       ShelleyMode{} -> encodePretty tip
-                       CardanoMode{} -> encodePretty tip
-        return output
+    let localNodeConnInfo = NewIPC.LocalNodeConnectInfo cModeParams network sockPath
+
+    tip <- liftIO $ NewIPC.getLocalChainTip localNodeConnInfo
+
+    let output = case NewIPC.localConsensusMode localNodeConnInfo of
+                   Mode.ByronMode   -> encodePretty tip
+                   Mode.ShelleyMode -> encodePretty tip
+                   Mode.CardanoMode -> encodePretty tip
```

Comment:
Do we really need to do this mode case analysis? The tip has a monomorphic type here. It does not depend on the node.

BTW, if the purpose of the JSON instance above being so human readable was because we want a human readable presentation here, then of course we can just do the human rendering here. JSON is primarily machine readable format.